receptorctl/receptorctl/socket_interface.py

```diff
@@ -124,12 +130,14 @@ def close(self):
             self.socket = None
 
     def connect_to_service(self, node, service, tlsclient):
+        self.connect()
         self.writestr(f"connect {node} {service} {tlsclient}\n")
         text = self.readstr()
         if not str.startswith(text, "Connecting"):
             raise RuntimeError(text)
 
     def submit_work(self, worktype, payload, node=None, tlsclient=None, ttl=None, params=None):
+        self.connect()
```

Comment:
If this method succeeds, it's going to leave the socket in an unusable state for further commands, so we need to note that the next command will need a fresh connection, probably by setting `self.socket = None`.

Also, we're returning the filelike object generated by `socket.makefile()`.  We want to leave this open so the caller can read from it, but we also want the caller to be able to close it.  The way `makefile` works is it duplicates the file descriptor, so in order to close the underlying socket connection, you have to close _both_ the socket _and_ the sockfile.  So I think we should go ahead and close the socket here, meaning that when the caller closes the sockfile, the connection will be dropped.